src/kOS.Safe/Compilation/Opcode.cs

```diff
@@ -1739,11 +1716,8 @@ public class OpcodeWait : Opcode
 
         public override void Execute(ICpu cpu)
         {
-            object waitTime = cpu.PopValue();
-            if (waitTime is double)
-                cpu.StartWait((double)waitTime);
-            else if (waitTime is int)
-                cpu.StartWait((int)waitTime);
+            object arg = cpu.PopValue();
+            cpu.StartWait(Convert.ToDouble(arg));
```

Comment:
We have methods over in FunctionBase that do safe numeric conversions (safe in that they will trap System casting exceptions and transform them into KOS exceptions instead, so users don't see the raw System exceptions when they pass bogus types of values to the functions.) 

It occurs to me that we might want to do a refactor later that adds that sort of thinking to all these kRISC opcode parameters.  Many of them are coming from the compiler directly injecting the ML code with the results of user expressions.